src/cmake/modules/GUDHI_third_party_libraries.cmake

```diff
@@ -52,14 +52,43 @@ option(WITH_GUDHI_USE_TBB "Build with Intel TBB parallelization" ON)
 
 # Find TBB package for parallel sort - not mandatory, just optional.
 if(WITH_GUDHI_USE_TBB)
-  set(TBB_FIND_QUIETLY ON)
-  find_package(TBB)
-  if (TBB_FOUND)
-    include(${TBB_USE_FILE})
-    message("TBB found in ${TBB_LIBRARY_DIRS}")
-    add_definitions(-DGUDHI_USE_TBB)
+  find_package(TBB CONFIG)
+  if(TARGET TBB::tbb)
+    # Specific windows case with its debug/release management
+    if(CMAKE_BUILD_TYPE MATCHES Debug)
```

Comment:
It is considered a Bad Thing™ to test CMAKE_BUILD_TYPE, some cmake generators (like visual studio) ignore it and produce build instructions for both release and debug. The advice is to try and use generator expressions like `$<$<CONFIG:Debug>:DEBUG_MODE>` where possible.
I think we can ignore this issue for now, but it is something to keep in mind.